### PR TITLE
feat(JOML): migrate BlockItemSystem and Physics#scanArea

### PR DESCRIPTION
**Pull request by pollend**
here is some work on migrating aabb for physics along with some effort to migrate BlockItemSystem.

### CODE REVIEW

File: engine/src/main/java/org/terasology/world/block/items/BlockItemSystem.java [hunk 1]
```diff
@@ -189,30 +190,33 @@ private boolean canPlaceBlock(Block block, Vector3i targetBlock, Vector3i blockP
...
+            blockBounds.maxY -= KinematicCharacterMover.VERTICAL_PENETRATION;
+            blockBounds.minZ += KinematicCharacterMover.HORIZONTAL_PENETRATION;
+            blockBounds.maxZ -= KinematicCharacterMover.HORIZONTAL_PENETRATION;
 
             /*
              * Calculations aren't exact and in the corner cases it is better to let the user place the block.
              */
-            min.add(ADDITIONAL_ALLOWED_PENETRATION, ADDITIONAL_ALLOWED_PENETRATION, ADDITIONAL_ALLOWED_PENETRATION);
-            max.sub(ADDITIONAL_ALLOWED_PENETRATION, ADDITIONAL_ALLOWED_PENETRATION, ADDITIONAL_ALLOWED_PENETRATION);
+            blockBounds.minX += ADDITIONAL_ALLOWED_PENETRATION;
```
Comment (DarkWeird) [inline engine/src/main/java/org/terasology/world/block/items/BlockItemSystem.java:210]: Seems there must be `-=`

File: engine/src/main/java/org/terasology/physics/bullet/BulletPhysics.java [hunk 2]
```diff
@@ -200,6 +206,30 @@ public short combineGroups(Iterable<CollisionGroup> groups) {
         return result;
     }
 
+    @Override
+    public List<EntityRef> scanArea(AABBf area, Iterable<CollisionGroup> collisionFilter) {
```
Comment (DarkWeird) [inline engine/src/main/java/org/terasology/physics/bullet/BulletPhysics.java:210]: Duplication with old method.

**Review by DarkWeird — CHANGES_REQUESTED** [review]

**Review by jdrueckert — COMMENTED** [review]
@pollend DarkWeird's review comments still seem to apply here.

**Comment by pollend** [comment]
I need to work on that.

**Review by DarkWeird — APPROVED** [review]

**Pull request merged**